lib/ddtrace/contrib/configuration/resolvers/regexp_resolver.rb

```diff
@@ -0,0 +1,31 @@
+require 'ddtrace/contrib/configuration/resolver'
+
+module Datadog
+  module Contrib
+    module Configuration
+      # Resolves a value to a configuration key
+      module Resolvers
+        # Matches strings against Regexps.
+        class RegexpResolver < Datadog::Contrib::Configuration::Resolver
+          def add_key(pattern)
+            patterns << pattern.is_a?(Regexp) ? pattern : /#{Regexp.quote(pattern)}/
```

Comment:
Although its kind of nice to convert everything to a Regex for consistency I think there's a couple of problems with interpolating strings to `Regexp`:

1. String values might match inadvertently, e.g. passing `'a'` would match `'a'` and `'aa'`.
2. There's an added expense to evaluating `Regexp` when doing `==` with a string might suffice.

I would recommend accepting any object, then in `#resolve` do a case statement such that if the pattern is a `Regexp`, do `=~`, otherwise coerce the pattern to a string and do `==`.